comp/core/workloadmeta/collectors/internal/kubemetadata/kubemetadata.go

```diff
@@ -229,10 +231,35 @@ func (c *collector) parsePods(
 			}
 		}
 
-		var nsLabels map[string]string
-		nsLabels, err = c.getNamespaceLabels(pod.Metadata.Namespace)
-		if err != nil {
-			log.Debugf("Could not fetch namespace labels for pod %s/%s: %v", pod.Metadata.Namespace, pod.Metadata.Name, err)
+		var nsLabels, nsAnnotations map[string]string
+
+		if c.dcaClient.Version().Major >= 7 && c.dcaClient.Version().Minor >= 55 {
+			// Cluster agent with version 7.55+
+			var nsMetadata *clusteragent.Metadata
+			nsMetadata, err = c.getNamespaceMetadata(pod.Metadata.Namespace)
+			if err != nil {
+				log.Errorf("Could not fetch namespace metadata for pod %s/%s: %v", pod.Metadata.Namespace, pod.Metadata.Name, err)
+			}
+
+			if nsMetadata != nil {
+				if c.collectNamespaceAnnotations {
+					nsAnnotations = nsMetadata.Annotations
+				}
+
+				if c.collectNamespaceLabels {
+					nsLabels = nsMetadata.Labels
+				}
+			}
+		} else {
+			// Cluster agent with version older than 7.55
+			nsLabels, err = c.getNamespaceLabels(pod.Metadata.Namespace)
+			if err != nil {
+				log.Errorf("Could not fetch namespace labels for pod %s/%s: %v", pod.Metadata.Namespace, pod.Metadata.Name, err)
+			}
+
+			if c.collectNamespaceAnnotations {
+				log.Errorf("Could not fetch namespace annotations for pod %s/%s: kubernetes_namespace_annotations_as_tags requires version 7.55 or later of the cluster agent", pod.Metadata.Namespace, pod.Metadata.Name)
```

Comment:
Not sure if this should be a warning instead. Error is the agent is normally used for more serious issues.

Reply:
TBH I also hesitated here. It depends on whether we consider this to be a serious issue or not. I think when this happens, it should be considered as unexpected behaviour, and the user should be alerted about it. 

But I don't have a strong opinion, I also agree that this doesn't stop the agent from running properly, but it results in missing something that the user would have expected to get.